### PR TITLE
Add random seed support for torch dropout

### PR DESCRIPTION
**Pull request by haifeng-jin**
As tested, there is no noticable overhead for the added conditions in the if statement.
It now can reproduce the same dropout with the same seed.

### CODE REVIEW

File: keras_core/random/seed_generator.py [hunk 1]
```diff
@@ -35,8 +35,10 @@ def __init__(self, seed, **kwargs):
         else:
             self.backend = backend
 
+        self.initialized_with_none = seed is None
```
Comment (fchollet) [inline keras_core/random/seed_generator.py:38]: Let's use something less verbose like `_initial_seed` (private). Then we can check `is not None`

File: keras_core/backend/torch/random.py [hunk 2]
```diff
@@ -126,7 +126,10 @@ def _get_concrete_noise_shape(inputs, noise_shape):
 
 
 def dropout(inputs, rate, noise_shape=None, seed=None):
-    if noise_shape is not None:
+    if noise_shape is not None or not (
+        (seed is None)
+        or (isinstance(seed, SeedGenerator) and seed.initialized_with_none)
+    ):
```
Comment (fchollet) [inline keras_core/backend/torch/random.py:132]: You can do plain chaining of the 3 `or`s

**Review by fchollet — COMMENTED** [review]
Thanks for the PR!

File: keras_core/backend/torch/random.py (hunk 2 above)
Comment (haifeng-jin) [reply keras_core/backend/torch/random.py:133]: `A or not (B or C)` is equivalent to `A or (not B and not C)`.
So I just chain them with `and` and `or`.
Seems not possible to chain them with all `or`s.

File: keras_core/random/seed_generator.py (hunk 1 above)
Comment (haifeng-jin) [reply keras_core/random/seed_generator.py]: Done.

**Review by fchollet — APPROVED** [review]
LGTM

**Pull request merged**